uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -974,12 +974,35 @@ public static function deleteAllNonExpiredTOILRequestsForAbsenceType($absenceTyp
    * @return boolean
    */
   public static function softDelete($id) {
-    $leaveRequest = self::findById($id);
+    $leaveRequest = self::findRecordById($id);
     $leaveRequest->is_deleted = 1;
     $leaveRequest->save();
   }
 
   /**
+   * This function is a variant of the LeaveRequest::findByID method.
+   * The difference is that it takes the is_deleted property into consideration when
+   * finding a leave request record.
+   *
+   * @param int $id
+   *   ID of leave request to be found
+   *
+   * @return \CRM_HRLeaveAndAbsences_BAO_LeaveRequest
+   *
+   * @throws \Exception
+   */
+  public static function findRecordById($id) {
```

Comment:
is the old findById still being used anywhere?

If you had to replace all the `findById` (for LeaveRequests) occurrences with `findRecordById`, then I think it would make more sense to override the `findById` method in the LeaveRequest BAO to only return non deleted leave requests